internal/proxy/task_search.go

```diff
@@ -297,15 +298,40 @@ func (t *searchTask) checkNq(ctx context.Context) (int64, error) {
 	return nq, nil
 }
 
-func setQueryInfoIfMvEnable(queryInfo *planpb.QueryInfo, t *searchTask) error {
+func setQueryInfoIfMvEnable(queryInfo *planpb.QueryInfo, t *searchTask, plan *planpb.PlanNode) error {
 	if t.enableMaterializedView {
 		partitionKeyFieldSchema, err := typeutil.GetPartitionKeyFieldSchema(t.schema.CollectionSchema)
 		if err != nil {
 			log.Warn("failed to get partition key field schema", zap.Error(err))
 			return err
 		}
 		if typeutil.IsFieldDataTypeSupportMaterializedView(partitionKeyFieldSchema) {
+			collInfo, colErr := globalMetaCache.GetCollectionInfo(t.ctx, t.request.GetDbName(), t.collectionName, t.CollectionID)
+			if colErr != nil {
+				log.Warn("failed to get collection info", zap.Error(colErr))
+				return err
+			}
+
+			iso, _, err := common.IsPartitionKeyIsolationPropEnabled(collInfo.properties)
```

Comment:
put isolation flag in collection info struct to avoid parsing every search request later